inference/serve.py

```diff
@@ -58,22 +58,22 @@ def get_deployed_models(args):
-    parser.add_argument("--model", default=None, type=str, help="model name or path")
-    parser.add_argument("--tokenizer", default=None, type=str, help="tokenizer name or path")
-    parser.add_argument("--port", default=8000, type=int, help="the port of deployment address")
-    parser.add_argument("--route_prefix", default=None, type=str, help="the route prefix for HTTP requests.")
-    parser.add_argument("--cpus_per_worker", default="24", type=int, help="cpus per worker")
-    parser.add_argument("--gpus_per_worker", default=0, type=float, help="gpus per worker, used when --device is cuda")
-    parser.add_argument("--hpus_per_worker", default=0, type=float, help="hpus per worker, used when --device is hpu")
-    parser.add_argument("--deepspeed", action='store_true', help="enable deepspeed inference")
-    parser.add_argument("--workers_per_group", default="2", type=int, help="workers per group, used with --deepspeed")
-    parser.add_argument("--ipex", action='store_true', help="enable ipex optimization")
-    parser.add_argument("--device", default="cpu", type=str, help="cpu, xpu, hpu or cuda")
-    parser.add_argument("--serve_local_only", action="store_true", help="only support local access to url")
-    parser.add_argument("--serve_simple", action="store_true", help="whether to serve OpenAI-compatible API for all models or serve simple endpoint based on model conf files.")
-    parser.add_argument("--keep_serve_terminal", action="store_true", help="whether to keep serve terminal.")
+    parser = argparse.ArgumentParser(description="Model Serve Script", add_help=True)
+    parser.add_argument("--config_file", type=str, help="Inference configuration file in YAML. If specified, all other arguments will be ignored.")
+    parser.add_argument("--model", default=None, type=str, help="Model name or path.")
+    parser.add_argument("--tokenizer", default=None, type=str, help="Tokenizer name or path.")
+    parser.add_argument("--port", default=8000, type=int, help="The port of deployment address.")
+    parser.add_argument("--route_prefix", default=None, type=str, help="The route prefix for HTTP requests.")
+    parser.add_argument("--cpus_per_worker", default="24", type=int, help="CPUs per worker.")
+    parser.add_argument("--gpus_per_worker", default=0, type=float, help="GPUs per worker, used when --device is cuda.")
+    parser.add_argument("--hpus_per_worker", default=0, type=float, help="HPUs per worker, used when --device is hpu.")
+    parser.add_argument("--deepspeed", action='store_true', help="Enable deepspeed inference.")
+    parser.add_argument("--workers_per_group", default="2", type=int, help="Workers per group, used with --deepspeed.")
+    parser.add_argument("--ipex", action='store_true', help="Enable ipex optimization.")
+    parser.add_argument("--device", default="cpu", type=str, help="CPU, XPU, HPU or CUDA.")
```

Comment:
Please keep uppercase and lowercase consistent here

Reply:
I think these are shortcuts and maybe should be uppercase.

Reply:
> I think these are shortcuts and maybe should be uppercase.

You need to check back to the code if this is used as device string and if the device string is case sensitive.

Reply:
Got it. I found some places use uppercase like [L60](https://github.com/intel/llm-on-ray/blob/87ffd4f92336e49fcb54882ff331b9cb6aa6ef79/finetune/finetune_config.py#L60C33-L60C33) and some use lowercase [L217](https://github.com/intel/llm-on-ray/blob/87ffd4f92336e49fcb54882ff331b9cb6aa6ef79/finetune/finetune.py#L217C8-L217C8)...

Reply:
Also changed default to uppercase, to see if somewhere will go wrong.

Reply:
> Also changed default to uppercase, to see if somewhere will go wrong.

Pls change default to lowercase as presented as torch device. 
https://pytorch.org/docs/stable/tensor_attributes.html#torch-device

Reply:
> Got it. I found some places use uppercase like [L60](https://github.com/intel/llm-on-ray/blob/87ffd4f92336e49fcb54882ff331b9cb6aa6ef79/finetune/finetune_config.py#L60C33-L60C33) and some use lowercase [L217](https://github.com/intel/llm-on-ray/blob/87ffd4f92336e49fcb54882ff331b9cb6aa6ef79/finetune/finetune.py#L217C8-L217C8)...

changed to lowercase as torch device, I would suggest use lowercase directly
https://github.com/intel/llm-on-ray/blob/87ffd4f92336e49fcb54882ff331b9cb6aa6ef79/finetune/finetune.py#L216

Reply:
Ok, changed to lowercase.

Reply:
please click resolve conversation above and below.